<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SideWidget</class>
 <widget class="QWidget" name="SideWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>659</width>
    <height>568</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>269</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QGroupBox" name="groupBox">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>0</y>
     <width>271</width>
     <height>231</height>
    </rect>
   </property>
   <property name="title">
    <string>Transformations</string>
   </property>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>3</x>
      <y>21</y>
      <width>261</width>
      <height>214</height>
     </rect>
    </property>
    <layout class="QFormLayout" name="formLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Rotate around x</string>
       </property>
       <property name="buddy">
        <cstring>rotate_x_slider</cstring>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QSlider" name="rotate_x_slider">
       <property name="minimum">
        <number>-180</number>
       </property>
       <property name="maximum">
        <number>180</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Rotate around y</string>
       </property>
       <property name="buddy">
        <cstring>rotate_y_slider</cstring>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QSlider" name="rotate_y_slider">
       <property name="minimum">
        <number>-180</number>
       </property>
       <property name="maximum">
        <number>180</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Rotate around z</string>
       </property>
       <property name="buddy">
        <cstring>rotate_z_slider</cstring>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QSlider" name="rotate_z_slider">
       <property name="minimum">
        <number>-180</number>
       </property>
       <property name="maximum">
        <number>180</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Zoom factor</string>
       </property>
       <property name="buddy">
        <cstring>zoom_factor_spin_box</cstring>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QDoubleSpinBox" name="zoom_factor_spin_box">
       <property name="toolTip">
        <string/>
       </property>
       <property name="decimals">
        <number>5</number>
       </property>
       <property name="minimum">
        <double>0.000100000000000</double>
       </property>
       <property name="maximum">
        <double>10000.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
       <property name="value">
        <double>1.000000000000000</double>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Translate along x</string>
       </property>
       <property name="buddy">
        <cstring>trans_x_spin_box</cstring>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QDoubleSpinBox" name="trans_x_spin_box">
       <property name="minimum">
        <double>-100.000000000000000</double>
       </property>
       <property name="maximum">
        <double>100.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Translate along y</string>
       </property>
       <property name="buddy">
        <cstring>trans_y_spin_box</cstring>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QDoubleSpinBox" name="trans_y_spin_box">
       <property name="minimum">
        <double>-100.000000000000000</double>
       </property>
       <property name="maximum">
        <double>100.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>Translate along z</string>
       </property>
       <property name="buddy">
        <cstring>trans_z_spin_box</cstring>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QDoubleSpinBox" name="trans_z_spin_box">
       <property name="minimum">
        <double>-100.000000000000000</double>
       </property>
       <property name="maximum">
        <double>100.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.100000000000000</double>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="label_10">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>390</y>
      <width>73</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>Scaling</string>
    </property>
    <property name="buddy">
     <cstring>zoom_factor_spin_box</cstring>
    </property>
   </widget>
  </widget>
  <widget class="QTabWidget" name="tabWidget">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>240</y>
     <width>631</width>
     <height>311</height>
    </rect>
   </property>
   <property name="currentIndex">
    <number>4</number>
   </property>
   <widget class="QWidget" name="tab_3">
    <attribute name="title">
     <string>Lab2</string>
    </attribute>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>103</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>Parametric curve:</string>
     </property>
     <property name="buddy">
      <cstring>trans_y_spin_box</cstring>
     </property>
    </widget>
    <widget class="QComboBox" name="Pc">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>10</y>
       <width>152</width>
       <height>24</height>
      </rect>
     </property>
    </widget>
    <widget class="QCheckBox" name="ShowTangent">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>251</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Show Tangent</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QCheckBox" name="ShowAcceleration">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>231</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Show Acceleration</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>73</width>
       <height>26</height>
      </rect>
     </property>
     <property name="text">
      <string>Scaling</string>
     </property>
     <property name="buddy">
      <cstring>zoom_factor_spin_box</cstring>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="scale_spin_box">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>100</y>
       <width>150</width>
       <height>26</height>
      </rect>
     </property>
     <property name="toolTip">
      <string/>
     </property>
     <property name="decimals">
      <number>5</number>
     </property>
     <property name="minimum">
      <double>0.000100000000000</double>
     </property>
     <property name="maximum">
      <double>10000.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.100000000000000</double>
     </property>
     <property name="value">
      <double>1.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>130</y>
       <width>101</width>
       <height>26</height>
      </rect>
     </property>
     <property name="text">
      <string>Div Point Count</string>
     </property>
     <property name="buddy">
      <cstring>zoom_factor_spin_box</cstring>
     </property>
    </widget>
    <widget class="QSlider" name="_div_point_count_widget">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>140</y>
       <width>160</width>
       <height>17</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>500</number>
     </property>
     <property name="value">
      <number>200</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="tab_4">
    <attribute name="title">
     <string>Lab3</string>
    </attribute>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>10</y>
       <width>181</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Selected cyclic curve:</string>
     </property>
    </widget>
    <widget class="QComboBox" name="selectedCyclicCurve">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>10</y>
       <width>76</width>
       <height>24</height>
      </rect>
     </property>
    </widget>
    <widget class="QCheckBox" name="showCyclicCurves">
     <property name="geometry">
      <rect>
       <x>330</x>
       <y>10</y>
       <width>191</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Show interpolating curves</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="showTangents">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>181</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Show tangents</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="showAccelerations">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>50</y>
       <width>181</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Show acceleration</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_13">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>90</y>
       <width>221</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Speed of object on selected curve</string>
     </property>
    </widget>
    <widget class="QSlider" name="speedOfSelectedCurve">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>90</y>
       <width>160</width>
       <height>17</height>
      </rect>
     </property>
     <property name="maximum">
      <number>5</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_14">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>120</y>
       <width>241</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Selected point on selected cyclic curve</string>
     </property>
    </widget>
    <widget class="QComboBox" name="selectedPoint">
     <property name="geometry">
      <rect>
       <x>260</x>
       <y>120</y>
       <width>76</width>
       <height>24</height>
      </rect>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="selectedX">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>150</y>
       <width>64</width>
       <height>26</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-99.000000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="selectedY">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>150</y>
       <width>64</width>
       <height>26</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-99.000000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="selectedZ">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>150</y>
       <width>64</width>
       <height>26</height>
      </rect>
     </property>
     <property name="minimum">
      <double>-99.000000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_15">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>150</y>
       <width>21</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>X:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_16">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>150</y>
       <width>21</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Y:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_17">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>150</y>
       <width>21</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Z:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_18">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>150</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Distance:</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="distance">
     <property name="geometry">
      <rect>
       <x>370</x>
       <y>150</y>
       <width>64</width>
       <height>26</height>
      </rect>
     </property>
     <property name="minimum">
      <double>0.000000000000000</double>
     </property>
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.010000000000000</double>
     </property>
     <property name="value">
      <double>0.500000000000000</double>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="tab">
    <property name="focusPolicy">
     <enum>Qt::WheelFocus</enum>
    </property>
    <attribute name="title">
     <string>Lab4</string>
    </attribute>
    <widget class="QComboBox" name="selectedSurface">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>30</y>
       <width>141</width>
       <height>24</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_19">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>101</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Selected surface</string>
     </property>
    </widget>
    <widget class="QComboBox" name="selectedTexture">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>70</y>
       <width>141</width>
       <height>24</height>
      </rect>
     </property>
    </widget>
    <widget class="QComboBox" name="selectedMaterial">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>110</y>
       <width>141</width>
       <height>24</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_20">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>70</y>
       <width>101</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Selected texture</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_21">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>110</y>
       <width>111</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Selected material</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="showTexture">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>150</y>
       <width>161</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Show texture</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="Lab5">
    <attribute name="title">
     <string>Lab5</string>
    </attribute>
    <widget class="QLabel" name="label_22">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>111</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Selected shader:</string>
     </property>
    </widget>
    <widget class="QComboBox" name="selecteShader">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>20</y>
       <width>191</width>
       <height>24</height>
      </rect>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="scale">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>50</y>
       <width>64</width>
       <height>26</height>
      </rect>
     </property>
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.050000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="smoothing">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>90</y>
       <width>64</width>
       <height>26</height>
      </rect>
     </property>
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.050000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="shading">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>130</y>
       <width>64</width>
       <height>26</height>
      </rect>
     </property>
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.050000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_23">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>60</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Scale factor</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_24">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>100</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Smoothing</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_25">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>140</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Shading</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="red">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>50</y>
       <width>64</width>
       <height>26</height>
      </rect>
     </property>
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.050000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="blue">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>90</y>
       <width>64</width>
       <height>26</height>
      </rect>
     </property>
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.050000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="green">
     <property name="geometry">
      <rect>
       <x>270</x>
       <y>130</y>
       <width>64</width>
       <height>26</height>
      </rect>
     </property>
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.050000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="label_26">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>50</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Red:</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_27">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>100</y>
       <width>71</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Blue</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_28">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>140</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Green</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="Patch">
    <attribute name="title">
     <string>Patch</string>
    </attribute>
    <widget class="QCheckBox" name="after_int">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>30</y>
       <width>111</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>domb</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="u_lines">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>30</y>
       <width>111</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>u</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="v_deriv">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>60</y>
       <width>111</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>dv</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="u_deriv">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>30</y>
       <width>111</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>du</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="v_lines">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>60</y>
       <width>111</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>v</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="patch_control_net">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>60</y>
       <width>91</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>control net</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="tab_2">
    <attribute name="title">
     <string>Arc</string>
    </attribute>
    <widget class="QCheckBox" name="arc_deriv1">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>10</y>
       <width>91</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>d1</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="arc_deriv2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>40</y>
       <width>91</width>
       <height>24</height>
      </rect>
     </property>
     <property name="text">
      <string>d2</string>
     </property>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
